packages/transport/src/create.ts

```diff
@@ -68,53 +71,59 @@ export const createChannelTransport = (
 
   let callbackError: Error | undefined;
 
-  let connected: boolean | undefined;
-
-  let connectionState: (a: boolean, r?: JsonValue) => void;
-  const providerAck = Promise.race([
-    new Promise<void>((_, reject) =>
-      setTimeout(
-        reject,
-        initTimeout,
-        new ConnectError('Channel connection timed out', ConnectErrorCode.Unavailable),
-      ),
-    ),
-    new Promise<void>(
-      (ack, reject) =>
-        (connectionState = (state: boolean, reason?: JsonValue) => {
-          connected = state;
-          if (connected) ack();
-          else {
-            port.close();
-            const err = errorFromJson(reason!, {}, new ConnectError('Connection rejected'));
-            reject(err);
-            callbackError ??= err;
-            throw err;
-          }
-        }),
-    ),
-  ]);
-
-  port.addEventListener('message', ev => {
-    try {
-      if (isTransportState(ev.data)) connectionState(ev.data.connected, ev.data.reason);
-      else if (isTransportEvent(ev.data)) {
-        const respond = pending.get(ev.data.requestId);
-        if (!respond) throw Error(`Request ${ev.data.requestId} not pending`);
-        respond(ev.data);
-      } else if (isTransportError(ev.data))
-        throw errorFromJson(ev.data.error, {}, new ConnectError('Transport error'));
-      else throw Error('Unknown transport item');
-    } catch (error) {
-      const err = ConnectError.from(error);
-      callbackError ??= err;
-      throw err;
-    }
-  });
+  let port: MessagePort | undefined;
+
+  const connect = async () =>
+    getPort(s.typeName).then((port: MessagePort) => {
```

Comment:
Can we use async/await here? Or actually, if it's a syncronous function, we can just call it.

Reply:
it's important to delay calling until first request, that's why it's wrapped in a function

Reply:
previously, just calling here was what caused tests to fail static import, even when the function was async

calling on creation caused init handshake at creation. the tests were noticing the rejected promise that hadn't surfaced to the caller

so wrapping it with this function and then calling from request with the conditional ensured it wasn't firing on actual creation

Reply:
My comment is less so about the wrapper `connect()` function. It's more so that we can do this:

```typescript
 const connect = async () => {
    const port = getPort(s.typeName);
    ...
```